05-UI-ReactJS/topic-3/src/App.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import Movies from './components/Movies';
+import AddMovie from './components/AddMovie';
+//import PropTypes from 'prop-types';
+
+import './App.css';
+
+class App extends React.Component {
+  constructor() {//defining initial keys
+    super();
+    this.state = {
+      movies: []
+    };
+  }
+//lifecicle method : triggerd when componenet re-render here comes ajax methods
+  componentWillMount() {
+    this.setState( { movies: [
+      {
+        title: 'Infinity war',
+        category: 'fiction',
+        year: '2018',
+        duration: '300'
+      },
+      {
+        title: 'Titanic',
+        category: 'drama',
+        year: '2018',
+        duration: '300'
+      },
+      {
+        title: 'Hangover',
+        category: 'comedy',
+        year: '2018',
+        duration: '300'
+      },
+    ]});
+  }
+  handleAddMovie(movie){
+    //data will not persist
+    let movies =this.state.movies;
+    movies.push(movie);
```

Comment:
This has an issue. Arrays and objects are being handled by referene. If you push the item into the object you will notice that it is changed in this.state before you do setState. You will need to clone the array, update it and use the new array as set state